### PR TITLE
feat: add role assign flow

### PR DESCRIPTION
**Pull request by howar31**
# Issue
- Add role assign function
- Add default role flow
- Add donation role flow

# Notice

# Dependency

### CODE REVIEW

File: configs/constants/constants.go [hunk 1]
```diff
@@ -26,4 +26,11 @@ const (
 	PrivilegeMember = 10
 	// PrivilegeAdmin ...
 	PrivilegeAdmin = 50
+
+	/* Role */
+	RoleIdExplorer    = 1
```
Comment (Aylie-Chou) [inline configs/constants/constants.go:31]: Question: why do we choose `id` rather than `key` to be the constants ?

Comment (howar31) [reply configs/constants/constants.go:31]: The `key` is a new column.  I'll merge the `key` related commit to this PR.

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
The code looks good with the role key !
Just have to check the details when assigning roles relationship. Please have a look at the comment.

File: configs/constants/constants.go (hunk 1 above)
Comment (howar31) [reply configs/constants/constants.go]: done

File: storage/user.go [hunk 2]
```diff
@@ -231,3 +243,24 @@ func (gs *GormStorage) UpdateReadPreferenceOfUser(userID string, readPreference
...
+	if err := gs.db.First(&role, "key = ?", roleKey).Error; err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to find role with Key: %s", roleKey))
+	}
+
+	// Remove all existing roles from the user
+	association := gs.db.Model(&user).Association("Roles")
+	association.Clear()
+
+	// Assign the new role to the user
+	association.Append(&role)
```
[the remark this answers is not in the document]
Comment (howar31) [reply storage/user.go]: done with Replace()

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**